### PR TITLE
 [2.3.2.r1.4] Nile sde support

### PR DESCRIPTION
**Pull request by opendata26**
This pr first adds initial snapdragon display engine support for nile with kgsl for 3d acceleration.

These are added as a separate dts and defconfig as they will need to be tested thoroughly before being merged.

### CODE REVIEW

**Comment by pablomh** [comment]
What kind of testing is needed?

File: drivers/gpu/drm/msm/sde/sde_hw_catalog.c [hunk 1]
```diff
@@ -3245,7 +3245,7 @@ static int _sde_hardware_pre_caps(struct sde_mdss_cfg *sde_cfg, uint32_t hw_rev)
 
 	rc = sde_hardware_format_caps(sde_cfg, hw_rev);
 
-	if (IS_MSM8996_TARGET(hw_rev)) {
+	if (IS_MSM8996_TARGET(hw_rev) || IS_SDM630_TARGET(hw_rev)) {
 		/* update msm8996 target here */
```
Comment (pablomh) [inline drivers/gpu/drm/msm/sde/sde_hw_catalog.c]: Update comment as well?

File: arch/arm64/boot/dts/qcom/sdm630-sde-overlay.dtsi [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	/delete-node/ qcom,mdss_wb_panel;
+
+	mdss_fb2: mdss_fb2 {
+	};
+};
+
+#include "sdm630-sde.dtsi"
+#include "sdm630-sde-display.dtsi"
+
+
```
Comment (pablomh) [inline arch/arm64/boot/dts/qcom/sdm630-sde-overlay.dtsi]: Redundant line.

**Comment by opendata26** [comment]
@pablomh a lot - the android user space doesn't even work yet, this can probably be default when tone-sde is stable.

EDIT: added with fixes

**Comment by oshmoun** [comment]
@opendata26 would the same be possible for loire too? or is there lack of hw support on that platform?

**Comment by opendata26** [comment]
@oshmoun it might be okay, you would have to try it to see though, message me on hangouts (ctatlor97@gmail.com) if you need any pointers

**Pull request merged**